copy_user_fields.py

```diff
@@ -0,0 +1,142 @@
+#! /usr/bin/python3
+"""
+Copy user defined fields and their mappings from a CloudV1 to a CloudV2 organization
+"""
+import argparse
+import itertools
+
+from client.cloud_v2 import *
+from client.cloud_v1 import *
+from client.fields import Fields
+
+
+def v1_get_unique_fields(fields: list) -> list:
+    def get_fields_by_name() -> dict:
+        v1_fields_by_name = [(field['name'], field) for field in fields]
+        # deduplicate
+        return dict([(field[0],
+                      [f[1] for f in v1_fields_by_name if f[0] == field[0]])
+                     for field in v1_fields_by_name])
+
+    def validate_same_config(fields: list) -> bool:
+        previous = fields[0]
+        for i in range(1, len(fields)):
+            current = fields[i]
+            if not (previous['fieldType'] == fields[i]['fieldType'] and
+                    previous['contentType'] == fields[i]['contentType'] and
+                    previous['fieldQueries'] == fields[i]['fieldQueries'] and
+                    previous['freeTextQueries'] == fields[i]['freeTextQueries'] and
+                    previous['facet'] == fields[i]['facet'] and
+                    previous['multivalueFacet'] == fields[i]['multivalueFacet'] and
+                    previous['sort'] == fields[i]['sort'] and
+                    previous['displayField'] == fields[i]['displayField']):
+                print(
+                    f'SKIPPING FIELD {current["name"]}. Found fields in CloudV1 with the same name but different configurations: {fields}')
+                return False
+            previous = fields[i]
+        return True
+
+    fields_by_name = get_fields_by_name()
+    unique_fields_by_name = \
+        [t for t in
+         [(field, fields_by_name[field])
+          if validate_same_config(fields_by_name[field]) else None
+          for field in fields_by_name]
+         if t is not None]
```

Comment:
Why not just do a empty list with a unique for that append to it (like in line #56)?